src/main/java/kg/jarkyn/cobspecserver/RequestParser.java

```diff
@@ -1,9 +1,61 @@
 package kg.jarkyn.cobspecserver;
 
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
 
 public class RequestParser {
+    private RequestLineParser requestLineParser = new RequestLineParser();
+
     public Request parse(InputStream inputStream) {
-        return null;
+        BufferedReader reader = inputToReader(inputStream);
+        String requestLine = extractRequestLine(reader);
+        return new Request(extractMethod(requestLine),
+                           extractPath(requestLine));
+    }
+
+    private String extractMethod(String requestLine) {
+        return requestLineParser.extractMethod(requestLine);
+    }
+
+    private String extractPath(String requestLine) {
+        return requestLineParser.extractPath(requestLine);
+    }
+
+    private String extractRequestLine(BufferedReader reader) {
+        try {
+            return reader.readLine();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private BufferedReader inputToReader(InputStream inputStream) {
+        InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+        return new BufferedReader(inputStreamReader);
+    }
+
+    private class RequestLineParser {
+
+        private final int FIRST_SEGMENT = 0;
```

Comment:
I do understand the argument against magic numbers, but these constant names aren't adding any meaning to me. Every programmer knows that indexing at 0 takes the first element of an array. I feel like you can remove the constants without losing expressiveness or meaning.